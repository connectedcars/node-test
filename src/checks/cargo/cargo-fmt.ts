import { filterDuplicates } from '../../common'
import { CheckAnnotation, CheckRunCompleted } from '../checks-common'
import { CargoFmtFile } from './cargo-types'

export interface CargoFmtInput {
  sha: string
  data: CargoFmtFile[]
}

export function cargoFmtCheck({ data, sha }: CargoFmtInput): CheckRunCompleted {
  if (Array.isArray(data)) {
    let annotations: CheckAnnotation[] = []
    for (const file of data) {
      for (const mismatch of file.mismatches) {
        const annotation: CheckAnnotation = {
          path: file.name,
          annotation_level: 'failure',
          start_line: mismatch.original_begin_line,
          end_line: mismatch.expected_end_line,
          message: `Original: ${mismatch.original}\n\nExpected: ${mismatch.expected}`,
          raw_details: JSON.stringify(mismatch, null, '    ')
        }
        annotations.push(annotation)
      }
    }
    if (annotations.length > 0) {
      annotations = filterDuplicates(annotations)

      const summary = `Total of ${annotations.length} ${annotations.length === 1 ? 'issue' : 'issues'}`
      return {
        name: 'cargo-fmt',
        head_sha: sha,
        conclusion: 'failure',
        status: 'completed',
        completed_at: new Date().toISOString(),
        output: {
          title: summary,
          summary,
          annotations
        }
      }
    }
    return {
      name: 'cargo-fmt',
      head_sha: sha,
      conclusion: 'success',
      status: 'completed',
      completed_at: new Date().toISOString(),
      output: {
        title: 'Found no issues',
        summary: 'Found no issues',
        annotations: []
      }
    }
  }
  return {
    name: 'cargo-fmt',
    head_sha: sha,
    conclusion: 'skipped',
    status: 'completed',
    completed_at: new Date().toISOString(),
    output: {
      title: 'Unexpected cargo fmt output',
      summary: ''
    }
  }
}
