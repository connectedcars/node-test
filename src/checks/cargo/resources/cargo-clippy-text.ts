import { CargoMessage } from '../cargo-types'

export const cargoClippyFailedOutput: CargoMessage[] = [
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered:
        'error: this comparison involving the minimum or maximum element for this type contains a case that is always true or always false\n   --> can/src/can_session.rs:391:12\n    |\n391 |         if 100 > i32::MAX {}\n    |            ^^^^^^^^^^^^^^\n    |\n    = note: `-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`\n    = help: because `i32::MAX` is the maximum value for this type, this comparison is always false\n    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons\n\n',
      children: [
        {
          children: [],
          code: null,
          level: 'note',
          message: '`-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message: 'because `i32::MAX` is the maximum value for this type, this comparison is always false',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message:
            'for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'clippy::absurd_extreme_comparisons',
        explanation: null
      },
      level: 'error',
      message:
        'this comparison involving the minimum or maximum element for this type contains a case that is always true or always false',
      spans: [
        {
          byte_end: 13338,
          byte_start: 13324,
          column_end: 26,
          column_start: 12,
          expansion: null,
          file_name: 'can/src/can_session.rs',
          is_primary: true,
          label: null,
          line_end: 391,
          line_start: 391,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 26,
              highlight_start: 12,
              text: '        if 100 > i32::MAX {}'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered: 'error: aborting due to previous error\n\n',
      children: [],
      code: null,
      level: 'error',
      message: 'aborting due to previous error',
      spans: []
    }
  }
]
