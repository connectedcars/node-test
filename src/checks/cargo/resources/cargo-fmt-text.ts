import { CargoFmtFile } from '../cargo-types'

export const cargoFmtFailedOutput: CargoFmtFile[] = [
  {
    name: '/checks-test/src/foo/bar.rs',
    mismatches: [
      {
        original_begin_line: 1,
        original_end_line: 7,
        expected_begin_line: 1,
        expected_end_line: 2,
        original: '\n\npub fn bar(\n\n)\n{\n    println!  (  "bar"  )  ;',
        expected: 'pub fn bar() {\n    println!("bar");'
      }
    ]
  },
  {
    name: '/checks-test/src/foo/mod.rs',
    mismatches: [
      {
        original_begin_line: 1,
        original_end_line: 5,
        expected_begin_line: 1,
        expected_end_line: 1,
        original: '\n\nmod bar\n;\n',
        expected: 'mod bar;'
      }
    ]
  },
  {
    name: '/checks-test/src/main.rs',
    mismatches: [
      {
        original_begin_line: 9,
        original_end_line: 9,
        expected_begin_line: 9,
        expected_end_line: 9,
        original: '',
        expected: 'mod foo;'
      },
      {
        original_begin_line: 10,
        original_end_line: 10,
        expected_begin_line: 11,
        expected_end_line: 12,
        original: ' mod foo  ;',
        expected: 'fn main() {\n    let foo = foo();'
      },
      {
        original_begin_line: 12,
        original_end_line: 16,
        expected_begin_line: 14,
        expected_end_line: 14,
        original: '\nfn main()\n{\n      let foo = foo();\n',
        expected: ''
      },
      {
        original_begin_line: 19,
        original_end_line: 19,
        expected_begin_line: 16,
        expected_end_line: 16,
        original: '',
        expected: ''
      },
      {
        original_begin_line: 27,
        original_end_line: 28,
        expected_begin_line: 23,
        expected_end_line: 23,
        original: '\n',
        expected: ''
      },
      {
        original_begin_line: 35,
        original_end_line: 38,
        expected_begin_line: 29,
        expected_end_line: 32,
        original: '\n\n    if { let b = false; b } {}\n',
        expected: '    if {\n        let b = false;\n        b\n    } {}'
      },
      {
        original_begin_line: 56,
        original_end_line: 56,
        expected_begin_line: 50,
        expected_end_line: 50,
        original: '    assert!(false);',
        expected: '        assert!(false);'
      },
      {
        original_begin_line: 71,
        original_end_line: 72,
        expected_begin_line: 65,
        expected_end_line: 65,
        original: '\n',
        expected: ''
      },
      {
        original_begin_line: 74,
        original_end_line: 75,
        expected_begin_line: 66,
        expected_end_line: 66,
        original: '\n',
        expected: ''
      }
    ]
  }
]
