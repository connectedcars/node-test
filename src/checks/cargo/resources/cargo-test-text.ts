import { CargoMessage } from '../cargo-types'

export const cargoTestFailedOutput: CargoMessage[] = [
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered:
        'error: expected one of `.`, `;`, `?`, `}`, or an operator, found `assert_eq`\n   --> can/src/can_session.rs:536:9\n    |\n535 |         assert!(false) \n    |                       - expected one of `.`, `;`, `?`, `}`, or an operator\n536 |         assert_eq!(session.typ, SessionType::TP2(0x89));\n    |         ^^^^^^^^^ unexpected token\n\n',
      children: [],
      code: null,
      level: 'error',
      message: 'expected one of `.`, `;`, `?`, `}`, or an operator, found `assert_eq`',
      spans: [
        {
          byte_end: 17846,
          byte_start: 17846,
          column_end: 23,
          column_start: 23,
          expansion: null,
          file_name: 'can/src/can_session.rs',
          is_primary: false,
          label: 'expected one of `.`, `;`, `?`, `}`, or an operator',
          line_end: 535,
          line_start: 535,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 23,
              highlight_start: 23,
              text: '        assert!(false) '
            }
          ]
        },
        {
          byte_end: 17865,
          byte_start: 17856,
          column_end: 18,
          column_start: 9,
          expansion: null,
          file_name: 'can/src/can_session.rs',
          is_primary: true,
          label: 'unexpected token',
          line_end: 536,
          line_start: 536,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 18,
              highlight_start: 9,
              text: '        assert_eq!(session.typ, SessionType::TP2(0x89));'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered: 'error: aborting due to previous error; 2 warnings emitted\n\n',
      children: [],
      code: null,
      level: 'error',
      message: 'aborting due to previous error; 2 warnings emitted',
      spans: []
    }
  },
  {
    reason: 'build-finished',
    success: false
  },
  {
    type: 'test',
    name: 'can_session::tests::tp2_stays_alive_after_applayer_error',
    event: 'failed',
    stdout:
      "thread 'can_session::tests::tp2_stays_alive_after_applayer_error' panicked at 'assertion failed: false', can/src/can_session.rs:535:9\nnote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n"
  },
  {
    type: 'test',
    event: 'started',
    name: 'passive_listeners::tester_detector::tests::obd2'
  },
  {
    type: 'test',
    name: 'passive_listeners::tester_detector::tests::detects_frames_on_diagnostic_relevant_arbitration_ids_sent_by_someone_else',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'passive_listeners::tester_detector::tests::obd2_but_not_functionally_addressed'
  },
  {
    type: 'test',
    name: 'passive_listeners::tester_detector::tests::doesnt_hardcode_obd2',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'server::tests::it_deserialize_null_sessions'
  },
  {
    type: 'test',
    name: 'passive_listeners::tester_detector::tests::functionally_addressed_but_not_obd2',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'server::tests::it_deserialize_sessions_with_numbered_level'
  },
  {
    type: 'test',
    name: 'passive_listeners::tester_detector::tests::obd2',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'server::tests::it_fails_empty_sessions'
  },
  {
    type: 'test',
    name: 'passive_listeners::tester_detector::tests::obd2_but_not_functionally_addressed',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'server::tests::it_fails_malformed_ecu_formats'
  },
  {
    type: 'test',
    name: 'server::tests::it_deserialize_null_sessions',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'server::tests::it_fails_nonsense_sessions'
  },
  {
    type: 'test',
    name: 'server::tests::it_deserialize_sessions_with_numbered_level',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'server::tests::it_works_on_wellformed_ecus'
  },
  {
    type: 'test',
    name: 'server::tests::it_fails_empty_sessions',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::check_sn_according_to_fiat'
  },
  {
    type: 'test',
    name: 'server::tests::it_fails_malformed_ecu_formats',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::check_sn_according_to_iso'
  },
  {
    type: 'test',
    name: 'server::tests::it_fails_nonsense_sessions',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_flow_control'
  },
  {
    type: 'test',
    name: 'server::tests::it_works_on_wellformed_ecus',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_frame_type_from_pci1'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::check_sn_according_to_fiat',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_handle_consecutive_frame'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::check_sn_according_to_iso',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_handle_first_frame'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_flow_control',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_handle_first_frame_unexpected'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_handle_consecutive_frame',
    event: 'ignored'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_handle_single_frame'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_frame_type_from_pci1',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_handle_single_frame_invalid_length'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_handle_first_frame',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_handle_single_frame_unexpected'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_handle_first_frame_unexpected',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_is_frame_for_us'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_handle_single_frame',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_in::test::test_waiting_for_first_or_single_frame_timeout'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_handle_single_frame_invalid_length',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_out::test::test_send'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_handle_single_frame_unexpected',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_out::test::test_send_not_ready'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_is_frame_for_us',
    event: 'ok'
  },
  {
    type: 'test',
    event: 'started',
    name: 'transport::isotp::transfer_out::test::test_single_frame'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_in::test::test_waiting_for_first_or_single_frame_timeout',
    event: 'ok'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_out::test::test_send',
    event: 'ok'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_out::test::test_send_not_ready',
    event: 'ok'
  },
  {
    type: 'test',
    name: 'transport::isotp::transfer_out::test::test_single_frame',
    event: 'ok'
  },
  {
    type: 'suite',
    event: 'failed',
    passed: 66,
    failed: 1,
    allowed_fail: 0,
    ignored: 1,
    measured: 0,
    filtered_out: 0
  }
]
