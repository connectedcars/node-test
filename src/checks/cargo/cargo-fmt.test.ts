import { cargoFmtCheck } from './cargo-fmt'
import { cargoFmtFailedOutput } from './resources/cargo-fmt-text'

describe('checks/cargo-fmt', () => {
  it('converts successful cargo fmt', () => {
    const result = cargoFmtCheck({ data: [], sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a' })
    expect(result).toMatchSnapshot({
      completed_at: expect.any(String)
    })
  })
  it('converts failed cargo fmt', () => {
    const result = cargoFmtCheck({ data: cargoFmtFailedOutput, sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a' })
    expect(result).toMatchSnapshot({
      completed_at: expect.any(String)
    })
  })
  it('converts buggy cargo fmt', () => {
    const result = cargoFmtCheck({
      data: {} as any,
      sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a'
    })
    expect(result).toMatchSnapshot({
      completed_at: expect.any(String)
    })
  })
})
