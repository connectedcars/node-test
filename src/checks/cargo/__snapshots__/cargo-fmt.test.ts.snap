// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks/cargo-fmt converts buggy cargo-fmt 1`] = `
Object {
  "completed_at": Any<String>,
  "conclusion": "skipped",
  "head_sha": "11963e3cb7ecbb9247f638cc0fb047173a62cf7a",
  "name": "cargo-fmt",
  "output": Object {
    "summary": "",
    "title": "Unexpected cargo fmt output",
  },
  "status": "completed",
}
`;

exports[`checks/cargo-fmt converts failed cargo-fmt 1`] = `
Object {
  "completed_at": Any<String>,
  "conclusion": "failure",
  "head_sha": "11963e3cb7ecbb9247f638cc0fb047173a62cf7a",
  "name": "cargo-fmt",
  "output": Object {
    "annotations": Array [
      Object {
        "annotation_level": "failure",
        "end_line": 2,
        "message": "Original: 

pub fn bar(

)
{
    println!  (  \\"bar\\"  )  ;

Expected: pub fn bar() {
    println!(\\"bar\\");",
        "path": "/checks-test/src/foo/bar.rs",
        "raw_details": "{
    \\"original_begin_line\\": 1,
    \\"original_end_line\\": 7,
    \\"expected_begin_line\\": 1,
    \\"expected_end_line\\": 2,
    \\"original\\": \\"\\\\n\\\\npub fn bar(\\\\n\\\\n)\\\\n{\\\\n    println!  (  \\\\\\"bar\\\\\\"  )  ;\\",
    \\"expected\\": \\"pub fn bar() {\\\\n    println!(\\\\\\"bar\\\\\\");\\"
}",
        "start_line": 1,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 1,
        "message": "Original: 

mod bar
;


Expected: mod bar;",
        "path": "/checks-test/src/foo/mod.rs",
        "raw_details": "{
    \\"original_begin_line\\": 1,
    \\"original_end_line\\": 5,
    \\"expected_begin_line\\": 1,
    \\"expected_end_line\\": 1,
    \\"original\\": \\"\\\\n\\\\nmod bar\\\\n;\\\\n\\",
    \\"expected\\": \\"mod bar;\\"
}",
        "start_line": 1,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 9,
        "message": "Original: 

Expected: mod foo;",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 9,
    \\"original_end_line\\": 9,
    \\"expected_begin_line\\": 9,
    \\"expected_end_line\\": 9,
    \\"original\\": \\"\\",
    \\"expected\\": \\"mod foo;\\"
}",
        "start_line": 9,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 12,
        "message": "Original:  mod foo  ;

Expected: fn main() {
    let foo = foo();",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 10,
    \\"original_end_line\\": 10,
    \\"expected_begin_line\\": 11,
    \\"expected_end_line\\": 12,
    \\"original\\": \\" mod foo  ;\\",
    \\"expected\\": \\"fn main() {\\\\n    let foo = foo();\\"
}",
        "start_line": 10,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 14,
        "message": "Original: 
fn main()
{
      let foo = foo();


Expected: ",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 12,
    \\"original_end_line\\": 16,
    \\"expected_begin_line\\": 14,
    \\"expected_end_line\\": 14,
    \\"original\\": \\"\\\\nfn main()\\\\n{\\\\n      let foo = foo();\\\\n\\",
    \\"expected\\": \\"\\"
}",
        "start_line": 12,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 16,
        "message": "Original: 

Expected: ",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 19,
    \\"original_end_line\\": 19,
    \\"expected_begin_line\\": 16,
    \\"expected_end_line\\": 16,
    \\"original\\": \\"\\",
    \\"expected\\": \\"\\"
}",
        "start_line": 19,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 23,
        "message": "Original: 


Expected: ",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 27,
    \\"original_end_line\\": 28,
    \\"expected_begin_line\\": 23,
    \\"expected_end_line\\": 23,
    \\"original\\": \\"\\\\n\\",
    \\"expected\\": \\"\\"
}",
        "start_line": 27,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 32,
        "message": "Original: 

    if { let b = false; b } {}


Expected:     if {
        let b = false;
        b
    } {}",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 35,
    \\"original_end_line\\": 38,
    \\"expected_begin_line\\": 29,
    \\"expected_end_line\\": 32,
    \\"original\\": \\"\\\\n\\\\n    if { let b = false; b } {}\\\\n\\",
    \\"expected\\": \\"    if {\\\\n        let b = false;\\\\n        b\\\\n    } {}\\"
}",
        "start_line": 35,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 50,
        "message": "Original:     assert!(false);

Expected:         assert!(false);",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 56,
    \\"original_end_line\\": 56,
    \\"expected_begin_line\\": 50,
    \\"expected_end_line\\": 50,
    \\"original\\": \\"    assert!(false);\\",
    \\"expected\\": \\"        assert!(false);\\"
}",
        "start_line": 56,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 65,
        "message": "Original: 


Expected: ",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 71,
    \\"original_end_line\\": 72,
    \\"expected_begin_line\\": 65,
    \\"expected_end_line\\": 65,
    \\"original\\": \\"\\\\n\\",
    \\"expected\\": \\"\\"
}",
        "start_line": 71,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 66,
        "message": "Original: 


Expected: ",
        "path": "/checks-test/src/main.rs",
        "raw_details": "{
    \\"original_begin_line\\": 74,
    \\"original_end_line\\": 75,
    \\"expected_begin_line\\": 66,
    \\"expected_end_line\\": 66,
    \\"original\\": \\"\\\\n\\",
    \\"expected\\": \\"\\"
}",
        "start_line": 74,
      },
    ],
    "summary": "Total of 11 issues",
    "title": "Total of 11 issues",
  },
  "status": "completed",
}
`;

exports[`checks/cargo-fmt converts successful cargo-fmt 1`] = `
Object {
  "completed_at": Any<String>,
  "conclusion": "success",
  "head_sha": "11963e3cb7ecbb9247f638cc0fb047173a62cf7a",
  "name": "cargo-fmt",
  "output": Object {
    "annotations": Array [],
    "summary": "Found no issues",
    "title": "Found no issues",
  },
  "status": "completed",
}
`;
